### PR TITLE
Multiplayer

### PR DESCRIPTION
**Pull request by FredTheDino**
You can now play online with your friends!

Too bad there's not much to play though...

Note the build commands that changed in #23 
```
cargo r -- --server  # Starts a server
cargo r -- --client  # Starts a client that connects to localhost
cargo r --           # Starts a game without any networking
```

### CODE REVIEW

File: game.sy [hunk 1]
```diff
@@ -412,13 +418,21 @@ start :: fn {
     // Added for safety - to avoid delta=0
     l_update'
 
-    
+    [0, 1] -> for_each' fn _:int {
```
Comment (sornas) [inline game.sy:421]: Lang question: do we want lang support for this? That is, "execute this function N times". Or is this a good enough workaround as written?

File: client.sy [hunk 2]
```diff
@@ -4,6 +4,14 @@ start :: fn {
     game.bind_keys'
 
     n_rpc_connect' "127.0.0.1"
+    n_rpc_server' game.player_join
```
Comment (sornas) [inline client.sy:7]: Maybe we could add a library function that blocks until we're connected to the server here?

File: game.sy [hunk 3]
```diff
@@ -252,28 +241,52 @@ init :: fn {
     }
 }
 
+num_players := 0
+player_join :: fn {
+    // TODO(ed): This is a bad idea - since we can
+    // have a race condition here.
+    // We probably want to respond to ONLY that client.
```
Comment (sornas) [inline game.sy:248]: +1 for this idea. Let's add this functionality if we find another place where responding to a single client directly is necessary. We already have a nice potential ID -- the index in the client handle vec.

File: game.sy (hunk 1 above)
Comment (FredTheDino) [reply game.sy:421]: We want something else. Maybe a "range" function? Writing out 100 numbers for example isn't feasible.

File: client.sy (hunk 2 above)
Comment (FredTheDino) [reply client.sy:7]: Then we can't close the game? I think it's better to have a much control as possible in the language. Maybe a callback is a more flexible idea?

**Pull request merged**